### PR TITLE
archlinux: Bunch of improvements

### PR DESCRIPTION
**Pull request by baalajimaestro**
- Reverted ccache to older version
- Ask for git username and email like in the ubuntu script

### CODE REVIEW

**Comment by msfjarvis** [comment]
shellcheck does not approve

**Comment by Yash-Garg** [comment]
Disable the checks at those lines @baalajimaestro kbye

File: setup/arch-manjaro.sh [hunk 1]
```diff
@@ -26,6 +26,25 @@ sudo chmod 644 /etc/udev/rules.d/51-android.rules
...
+echo "Reverting ccache to older version....."
+sudo rm -rf /usr/bin/ccache
+wget https://github.com/ccache/ccache/releases/download/v3.6/ccache-3.6.tar.xz
+tar -xvf ccache-3.6.tar.xz
+sudo pacman -S asciidoc -y
+cd ccache-3.6/
+export CC="clang"
+./autogen.sh
+./configure
+sed -i 's/-Wall -W/-Wall -W -Wno-implicit-fallthrough -Wno-extra-semi-stmt/g' Makefile
```
Comment (msfjarvis) [inline setup/arch-manjaro.sh]: Just pass this through `CFLAGS` and `CXXFLAGS` to `./configure`

File: setup/arch-manjaro.sh [hunk 2]
```diff
@@ -26,6 +26,43 @@ sudo chmod 644 /etc/udev/rules.d/51-android.rules
 sudo chown root /etc/udev/rules.d/51-android.rules
 sudo udevadm control --reload-rules
 
+# Revert ccache to an older version.
+echo "Reverting ccache to older version....."
+sudo rm -rf /usr/bin/ccache
+wget https://github.com/ccache/ccache/releases/download/v3.6/ccache-3.6.tar.xz
+tar -xvf ccache-3.6.tar.xz
+sudo pacman -S asciidoc -y
+cd ccache-3.6/
```
Comment (msfjarvis) [inline setup/arch-manjaro.sh]:
```suggestion
cd ccache-3.6/ || exit
```

File: setup/arch-manjaro.sh [hunk 3]
```diff
@@ -26,6 +26,43 @@ sudo chmod 644 /etc/udev/rules.d/51-android.rules
...
+export CC="clang"
+./autogen.sh
+./configure
+sed -i 's/-Wall -W/-Wall -W -Wno-implicit-fallthrough -Wno-extra-semi-stmt/g' Makefile
+make
+sudo make install
+
+# Cleanup
+
+cd ..
```
Comment (msfjarvis) [inline setup/arch-manjaro.sh]:
```suggestion
cd .. || exit
```

**Comment by akhilnarang** [comment]
What's wrong with the default ccache version?

**Comment by baalajimaestro** [comment]
The current ccache is 3.7.5 and derps on building android 10, with the strcpy_a9 issue

**Comment by msfjarvis** [comment]
> The current ccache is 3.7.5 and derps on building android 10, with the strcpy_a9 issue

I don't see an [open issue](https://github.com/ccache/ccache/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+strcpy_a9) about it yet, when are you planning on doing that?

**Comment by akhilnarang** [comment]
A quick google search led to me [this](https://github.com/sonyxperiadev/bug_tracker/issues/492), and through that, [this](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=944399). (should open one on ccache's github repo like Harsh said, though.

**Comment by merothh** [comment]
probably better to downgrade the package using pacman ?

**Comment by merothh** [comment]
probably a better idea to install the older version through the archive repos and pacman. Assuming 3.6-1 is what we want. `sudo pacman -U https://archive.archlinux.org/packages/c/ccache-3.6-1-x86_64.pkg.tar.xz 
`

**Comment by akhilnarang** [comment]
Yeah that'll probably be a better idea, @DD3Boh had done something similar here a year ago, iirc for `curl`

**Comment by merothh** [comment]
Also don't think we need to go down as far as 3.6, will check and confirm

**Comment by merothh** [comment]
3.7-4 works fine for me.

**Comment by xstefen** [comment]
> 3.7-4 works fine for me.

Ive tried every version in the Arch archive with the same error each time. With clean ccache and clean build. Did I miss a step lol? https://del.dog/lusivogaru

**Comment by merothh** [comment]
> > 3.7-4 works fine for me.
> 
> Ive tried every version in the Arch archive with the same error each time. With clean ccache and clean build. Did I miss a step lol? https://del.dog/lusivogaru

I just had to clean my out folder.

**Comment by jrosdahl** [comment]
Fixed in ccache v3.7.6, see https://github.com/ccache/ccache/issues/489.

**Comment by msfjarvis** [comment]
Closing since ccache bug is resolved and other review comments were not addressed.

**Pull request closed without merging**